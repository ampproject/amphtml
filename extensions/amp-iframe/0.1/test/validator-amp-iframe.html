<!--
  Test Description:
  Tests for the amp-iframe tag. See the inline comments.
-->
<!doctype html>
<html âš¡ lang="en">
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="./regular-html-version.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-latest.js"></script>
  <!-- Also include amp-bind to test special case interactions -->
  <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-latest.js"></script>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
</head>
<body>
  <!-- Valid: Basic tests -->
  <amp-iframe width="200" height="200" src="https://example.com"></amp-iframe>
  <amp-iframe width="200" height="200" srcdoc="<p>Hello World</p>"></amp-iframe>

  <!-- Valid: Allowed attributes tests -->
  <amp-iframe width="200" height="200" src="https://example.com" allow="geolocation; microphone" allowfullscreen allowpaymentrequest allowtransparency frameborder="0" referrerpolicy="origin" tabindex="-1" resizable sandbox="allow-same-origin allow-scripts" scrolling="no" ></amp-iframe>

  <!-- Valid: We allow the src attribute to be bindable -->
  <amp-iframe width="200" height="200" src="https://example.com" [src]="iframeurl"></amp-iframe>

  <!-- Invalid: We do not allow to bind to the src attribute, and also specify a srcdoc value -->
  <amp-iframe width="200" height="200" srcdoc="<p>Hello World</p>" [src]="iframeurl"></amp-iframe>


  <!-- Mandatory amp-state to avoid amp-bind validator errors -->
  <amp-state id=myState>
    <script type="application/json">
      {"iframeurl": "https://example.com"}
    </script>
  </amp-state>


</body>
</html>
