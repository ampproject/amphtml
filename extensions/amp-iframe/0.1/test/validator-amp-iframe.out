FAIL
|  <!--
|    Copyright 2015 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    Tests for the amp-iframe tag. See the inline comments.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|    <meta charset="utf-8">
|    <link rel="canonical" href="./regular-html-version.html">
|    <meta name="viewport" content="width=device-width,minimum-scale=1">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|    <script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-latest.js"></script>
|    <!-- Also include amp-bind to test special case interactions -->
|    <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-latest.js"></script>
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|  </head>
|  <body>
|    <!-- Valid: Basic tests -->
|    <amp-iframe width="200" height="200" src="https://example.com"></amp-iframe>
|    <amp-iframe width="200" height="200" srcdoc="<p>Hello World</p>"></amp-iframe>
|
|    <!-- Valid: Allowed attributes tests -->
|    <amp-iframe width="200" height="200" src="https://example.com" allow="geolocation; microphone" allowfullscreen allowpaymentrequest allowtransparency frameborder="0" referrerpolicy="origin" resizable sandbox="allow-same-origin allow-scripts" scrolling="no" ></amp-iframe>
|
|    <!-- Valid: We allow the src attribute to be bindable -->
|    <amp-iframe width="200" height="200" src="https://example.com" [src]="iframeurl"></amp-iframe>
|
|    <!-- Invalid: We do not allow to bind to the src attribute, and also specify a srcdoc value -->
|    <amp-iframe width="200" height="200" srcdoc="<p>Hello World</p>" [src]="iframeurl"></amp-iframe>
>>   ^~~~~~~~~
amp-iframe/0.1/test/validator-amp-iframe.html:44:2 The attribute 'src' in tag 'amp-iframe' is missing or incorrect, but required by attribute '[src]'. (see https://www.ampproject.org/docs/reference/components/amp-iframe) [DISALLOWED_HTML]
|
|
|    <!-- Mandatory amp-state to avoid amp-bind validator errors -->
|    <amp-state id=myState>
|      <script type="application/json">
|        {"iframeurl": "https://example.com"}
|      </script>
|    </amp-state>
|
|
|  </body>
|  </html>
